### PR TITLE
Timer updates: add iterations option, warmups, print split timers as csv to stdout

### PR DESCRIPTION
**Pull request by jhdavis8**

### CODE REVIEW

File: .dir-locals.el
Comment (joy-kitson) [inline .dir-locals.el:11]: Is there a vim equivalent of this? Might be good to have if non-emacs users are planning to edit the code, since this seems like an easy thing to mess up without realising

File: cuda/Simulation.cu
Comment (joy-kitson) [inline cuda/Simulation.cu]: Spelling out `host_to_device` might be clearer than using an abbreviated name (particularly if we want this to eventually get merged in upstream)

File: cuda/Simulation.cu
Comment (joy-kitson) [inline cuda/Simulation.cu:30]: Specifying the warmup iterations at runtime seems like it'd be more robust than hard-coding a fixed ratio

File: sycl/io.cpp
Comment (joy-kitson) [inline sycl/io.cpp:89]: This looks like it should be on two separate lines

**Review by joy-kitson — APPROVED** [review]
Overall this looks good, although I've noted a few minor concerns. Additionally, there seems to be a reasonable amount of repeated code (ex: refining the profile struct separately for each model). I know this mostly comes from inheriting the existing code structure, but it'd seem a cleaner to put the repeated definitions in a short header file at the top level and just include it.

File: cuda/Simulation.cu
Comment (pranav-sivaraman) [inline cuda/Simulation.cu]: This seems only be getting the final kernel time? I thought we wanted to keep track of all kernel time iterations?

File: cuda/Main.cu
Comment (pranav-sivaraman) [inline cuda/Main.cu]: If we want to keep track of the total time for all the kernel iterations we should also add a column indicating the number of iterations to easily calculate the average later.

File: cuda/Simulation.cu
Comment (jhdavis8) [reply cuda/Simulation.cu]: Will do

File: cuda/Simulation.cu
Comment (jhdavis8) [reply cuda/Simulation.cu:30]: Will do

File: cuda/Main.cu
Comment (jhdavis8) [reply cuda/Main.cu]: Will do

**Review by pranav-sivaraman — APPROVED** [review]
Looks Good! All models building on Perlmutter with a few minor changes to the CMake. I'll let you review them before you can merge.

**Comment by jhdavis8** [comment]
@joy-kitson just want to wait to get your new review before we merge this.

**Pull request merged**
